doc/examples/segmentation/plot_chanvese.py

```diff
@@ -0,0 +1,73 @@
+""" 
+======================
+Chan-Vese Segmentation
+======================
+
+The Chan-Vese Algorithm is an algorithm designed to segmentent objects
+without clearly defined boundaries. This algorithm is based on level
+sets and that are evolved iteatively to minimize an energy, which is
+defined by weighed values corresponding to the sum of differences in
+intensity from the average value outside the segmented region, the sum
+of differences from the average value inside the segmented region, and
+a term which is dependant on the length of the boundary of the
+segmented region.
+
+This algorithm was first proposed by Tony Chan and Luminita Vese, in 
+an publicaion entitled "An Active Countour Model Without Edges" [1].
+
+This implementation of the algorithm is somewhat simplified.
+The arguments are as follows:
+
+lambda1 : weight for the difference from the average 'inside' the
+          segmented region.
+lambda2 : weight for the difference from the average 'outside' the
+          segmented region.
+mu : weight for the term that is dependant on the length of the
+     boundary of the segmented region
+tol : this defines the stop condition for the Algorithm. A lower value
+      means more iterations and higher accuracy.
+maxiter : maximum number of iterations before the algorithm stops. This
+          is used to avoid endless loops if the solution is divergent.
+dt: a multiplication factor used to speed up the algorithm. Increasing
+    it may cause convergence problems.
+    
+Typical values for lambda1 and lambda2 are 1. If the 'background' is
+very different from the segmented objet in terms of distribution (for
+example, a uniform black image with figures of varying intensity), then
+these values should be different from each other.
+
+Typical values for mu are between 0 and 1, though higher values can be
+used when dealing with shapes with very ill-defined contours.
+
+The Algorithm also returns a list of values which coorespond to the
+energy at each iteration. This can be used to adjust the various
+parameters described above.
+
+References
+----------
+.. [1] An Active Contour Model without Edges, Tony Chan and 
+       Luminita Vese, Scale-Space Theories in Computer Vision, 1999
+.. [2] Chan-Vese Segmentation, Pascal Getreuer Image Processing On
+       Line, 2 (2012), pp. 214-224.
+.. [3] The Chan-Vese Algorithm - Project Report, Rami Cohen
+       http://arxiv.org/abs/1107.2782 , 2011
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+from skimage.data import camera
+from skimage.segmentation.chanvese import chan_vese
+
+
+U0 = camera().astype('float')/255.
+# Feel free to play around with these parameters to see how they impact the
+# result
+cv = chan_vese(U0, mu=0.5, lambda1=1, lambda2=0.5, tol=1, maxiter=100, dt=100)
+
+fig, ax = plt.subplots(2, 1)
+ax[0].imshow(cv[0])
+ax[0].set_title("Chan-Vese segmentation - "+str(len(cv[1]))+" iterations.")
```

Comment:
Could be a nice idea to `ax[0].set_axis_off()`, as the ticklabels aren't really informative.